bulk-model-sync-lib/src/jvmTest/kotlin/org/modelix/model/sync/bulk/ModelImporterOrderPropertyTest.kt

```diff
@@ -0,0 +1,355 @@
+            }
+        }
+    }
+
+    init {
+        TestLanguage.register()
+    }
+
+    private fun loadModelIntoBranch(modelData: ModelData): IBranch {
+        val store = ObjectStoreCache(MapBasedStore())
+        val tree = CLTree(
+            data = null,
+            repositoryId_ = null,
+            store_ = store,
+            useRoleIds = true,
+        )
+        val idGenerator = IdGenerator.getInstance(1)
+        val pBranch = PBranch(tree, idGenerator)
+
+        pBranch.runWrite {
+            modelData.load(pBranch)
+        }
+
+        return pBranch.withAutoTransactions()
+    }
+
+    @Test
+    fun `unordered children are not reorder`() {
+        // language=json
+        val dataBeforeImportJson = """
+        {
+          "root": {
+            "id": "root1",
+            "children": [
+              {
+                "id": "parent1",
+                "concept": "uid1",
+                "children": [
+                  {
+                    "id": "child1",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  },
+                  {
+                    "id": "child2",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  },
+                  {
+                    "id": "child3",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  }
+                ]
+              }
+            ]
+          }
+        }
+        """.trimIndent()
+        val dataBeforeImport = ModelData.fromJson(dataBeforeImportJson)
+        val branch = loadModelIntoBranch(dataBeforeImport)
+        val modelImporter = ModelImporter(branch.getRootNode())
+
+        // language=json
+        val dataToImportJson = """
+        {
+          "root": {
+            "id": "root1",
+            "children": [
+              {
+                "id": "parent1",
+                "concept": "uid1",
+                "children": [
+                  {
+                    "id": "child2",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  },
+                  {
+                    "id": "child4",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  },
+                  {
+                    "id": "child1",
+                    "concept": "uid1",
+                    "role": "uid2"
+                  }
+                ]
+              }
+            ]
+          }
+        }
+        """.trimIndent()
+        val dataToImport = ModelData.fromJson(dataToImportJson)
+        modelImporter.import(dataToImport)
+
+        val children = branch.getRootNode().allChildren.first().allChildren.toList()
+        // child1 and child2 should not be reordered because they have an unordered role
+        // child3 is removed
+        // child4 is added at the end
+        assertEquals(listOf("child1", "child2", "child4"), children.map(INode::originalId))
```

Comment:
If `IChildLink.isOrdered` is false then we can't expect any specific order. `assertEquals(setOf("child1", "child2", "child4"), children.map(INode::originalId).toSet())` would be correct.